### PR TITLE
Best support Internet Explorer

### PR DESCRIPTION
**Pull request by gchoqueux**
- Move header's shader in basic material and  fix error LogDepth in IE
- PolyFill Event for IE

### CODE REVIEW

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js [hunk 1]
```diff
@@ -13,11 +14,11 @@ import TileProvider from 'Core/Commander/Providers/TileProvider';
 import CoordCarto from 'Core/Geographic/CoordCarto';
 import Ellipsoid from 'Core/Math/Ellipsoid';
 import Projection from 'Core/Geographic/Projection';
+import 'Core/Commander/Interfaces/Event';
```
Comment (tbroyer) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: I'd rather add the polyfills to the WebPack entry-point, like we did for the others (fetch and promises)

File: src/Renderer/GlobeDepthMaterial.js [hunk 2]
```diff
@@ -6,8 +6,8 @@ var GlobeDepthMaterial = function(otherMaterial) {
 
     BasicMaterial.call(this);
 
-    this.vertexShader = GlobeDepthVS;
-    this.fragmentShader = GlobeDepthFS;
+	this.vertexShader = this.vertexShaderHeader + GlobeDepthVS;
+	this.fragmentShader = this.fragmentShaderHeader + GlobeDepthFS;
```
Comment (tbroyer) [inline src/Renderer/GlobeDepthMaterial.js]: Indentation change looks like a mistake

File: src/Renderer/LayeredMaterial.js [hunk 3]
```diff
@@ -45,42 +45,48 @@ var LayeredMaterial = function(id) {
 
     BasicMaterial.call(this);
 
-    var maxTexturesUnits = gfxEngine().glParams.maxTexturesUnits;
+    var maxTexturesUnits =  gfxEngine().glParams.maxTexturesUnits;
     this.vertexShader = GlobeVS;
+	var nbSamplers = Math.min(maxTexturesUnits-1,16-1);
```
Comment (tbroyer) [inline src/Renderer/LayeredMaterial.js]: nit: indentation

File: src/Renderer/LayeredMaterial.js [hunk 4]
```diff
@@ -45,42 +45,48 @@ var LayeredMaterial = function(id) {
...
+    this.fragmentShaderHeader += getColorAtIdUv(nbSamplers);
 
-    if (showDebug)
-        customFS += '#define DEBUG\n';
-
-    customFS += getColorAtIdUv(nbSamplers);
-
-    this.fragmentShader = customFS + GlobeFS;
+    this.fragmentShader = this.fragmentShaderHeader + GlobeFS;
+	this.vertexShader = this.vertexShaderHeader + GlobeVS;
```
Comment (tbroyer) [inline src/Renderer/LayeredMaterial.js]: nit: identation

File: src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js (hunk 1 above)
Comment (gchoqueux) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: Where?

Comment (gchoqueux) [inline src/Core/Commander/Interfaces/ApiInterface/ApiGlobe.js]: in webPack config?
Which package use?

**Comment by gchoqueux** [comment]
r?

File: src/Core/Commander/Interfaces/Event.js [hunk 5]
```diff
@@ -1,34 +1,23 @@
...
-
-
-/**
- */
-Event.prototype.xmlToEvents = function() {
-    //TODO: Implement Me
-
-};
+    // PolyFill Event to IE
+    var CustomEvent = function ( event, params )
```
Comment (peppsac) [inline src/Core/Commander/Interfaces/Event.js]: Is this file still needed?

**Comment by gchoqueux** [comment]
Branche is updated

**Comment by gchoqueux** [comment]
r?

File: src/Renderer/LayeredMaterial.js [hunk 6]
```diff
@@ -45,42 +45,48 @@ var LayeredMaterial = function(id) {
...
+        for(var i=0;i<array.length;i++)
+            array[i] = remp;
+    };
+
     // Uniform three js needs no empty array
     for (var l = 0; l < nbLayer; l++) {
 
         // WARNING TODO prevent empty slot, but it's not the solution
-        this.pitScale[l] = Array(nbSamplers).fill(vector);
+        this.pitScale[l] = Array(nbSamplers);//.fill(vector);
```
Comment (peppsac) [inline src/Renderer/LayeredMaterial.js]: Remove dead code and add a comment stating why we can't use `Array.fill`

**Comment by peppsac** [comment]
LGTM merging
